### PR TITLE
Add some check for fs-version-sudo

### PR DESCRIPTION
**Pull request by Lambourl**
The purpose of this Pr is to: 
- Check if there is enough space on the fs to install tipi
- Check if the version of ubuntu is at least 20.04 
- Check if sudo is present

### CODE REVIEW

**Comment by Lambourl** [comment]
The test on the docker does not work, because the cli does not check the presence or not of sudo

**Review by pysco68 — CHANGES_REQUESTED** [review]
The sudo part looks relatively sound, but the os check is def. lacking a check for the distro and some safety in order for it to be useful. 

I'd say fix that and remove the disk space thing for the time being.

**Comment by Lambourl** [comment]
Ok I will remove the disk size check and put the bone check back in a cleaner way

File: install/install_for_macos_linux.sh [hunk 1]
```diff
@@ -1,14 +1,19 @@
 #!/bin/bash
 
 VERSION="${TIPI_INSTALL_VERSION:-v0.0.28}"
+
  if [ "$(uname)" == "Linux" ]; then
     TIPI_URL="https://github.com/tipi-build/cli/releases/download/$VERSION/tipi-$VERSION-linux-x86_64.zip" 
+    AVAIABLE_SIZE_FS=$(df -H /dev/sda1  |  awk '{ print $4}' | cut -d'G' -f1 | cut -d'e' -f2)
```
[the remark this answers is not in the document]
Comment (Lambourl) [reply install/install_for_macos_linux.sh]: but it's still stored in /dev/sda* right?

File: install/install_for_macos_linux.sh [hunk 2]
```diff
@@ -18,12 +23,23 @@ info() {
...
+  printf "\e[1;33m ---> $1 \e[0m \n"
+}
+
+sucess() {
+  printf "\e[1;32m ---> $1 \e[0m \n"
 }
 
+if [[ -n "$UBUNTU_VERSION" ]] && [ "$UBUNTU_VERSION" -lt 20 ] ; then
+  abort "Minimum version supported by tipi is ubuntu 20.04"
+fi
```
[the remark this answers is not in the document]
Comment (Lambourl) [reply install/install_for_macos_linux.sh]: My bad

File: install/install_for_macos_linux.sh [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
+  printf "\e[1;33m ---> $1 \e[0m \n"
+}
+
  if [ "$(uname)" == "Linux" ]; then
     TIPI_URL="https://github.com/tipi-build/cli/releases/download/$VERSION/tipi-$VERSION-linux-x86_64.zip" 
+    AVAIABLE_SIZE_FS=$(df -H $HOME_DIR | awk '{ print $4}'  | cut -d'G' -f1 | cut -d'l' -f2 | tr -d '\n' )
+    DISTRO_NAME=$(cat /etc/*elease | grep DISTRIB_ID | cut -d '=' -f2)
+    DISTRO_VERSION=$(cat /etc/*elease | grep DISTRIB_RELEASE | cut -d '=' -f2 | sed 's@^[^0-9]*\([0-9]\+\).*@\1@')
+    if [[ -z "$DISTRO_VERSION" ]] || [[ -z "$DISTRO_NAME" ]] || ( [ "$DISTRO_VERSION" -lt 20 ] && [ "$DISTRO_NAME" != "Ubuntu" ]);then
+      warning "tipi is designed to work with at least Ubuntu 20.04"
```
Comment (pysco68) [inline install/install_for_macos_linux.sh:17]: => tipi is currently supported on Ubuntu 20.04 or later only. You are running an unsuported distribution.

File: install/install_for_macos_linux.sh [hunk 4]
```diff
@@ -1,14 +1,31 @@
...
+      warning "But would you still like to try to install tipi (y/n) ?"
```
Comment (pysco68) [inline install/install_for_macos_linux.sh:18]: => Do you want to install tipi anyway? (y/n)

**Review by pysco68 — CHANGES_REQUESTED** [review]
LGTM except for the phrasing (see my comments)

File: install/install_for_macos_linux.sh [hunk 5]
```diff
@@ -32,24 +53,38 @@ if [ -f "/etc/arch-release" ]; then
...
+    return 1
+  fi
+}
+
+CURRENT_USER=$(whoami)
+SUDO_COMMAND=''
+if  command -v sudo &> /dev/null
+then
+    SUDO_COMMAND='sudo'
+fi
```
Comment (Tachi107) [inline install/install_for_macos_linux.sh:67]: When `sudo` isn't available, you could fall back to using [`doas`](https://wiki.debian.org/Doas) instead

Comment (daminetreg) [reply install/install_for_macos_linux.sh:67]: Cool tip do you know if it’s available on Arch per default?

Comment (daminetreg) [reply install/install_for_macos_linux.sh:67]: Thank you very much for pointing this tool.

Comment (Tachi107) [reply install/install_for_macos_linux.sh:67]: > Cool tip do you know if it’s available on Arch per default?

I don't know for sure, but I think it isn't. It's more popular in the BSD world, but is also used on Linux by people (like me) who prefer a smaller and safer alternative.

I would set `SUDO_COMMAND` to `sudo` if available, then fallback to `doas` if available, and then warn that no tool to elevate privileges was found. Oh, by the way, a lot of distros ship `pkexec` too, as part of the Polkit (and systemd) suite

File: install/install_for_macos_linux.sh [hunk 6]
```diff
@@ -14,11 +14,13 @@ warning() {
     DISTRO_NAME=$(cat /etc/*elease | grep DISTRIB_ID | cut -d '=' -f2)
     DISTRO_VERSION=$(cat /etc/*elease | grep DISTRIB_RELEASE | cut -d '=' -f2 | sed 's@^[^0-9]*\([0-9]\+\).*@\1@')
     if [[ -z "$DISTRO_VERSION" ]] || [[ -z "$DISTRO_NAME" ]] || ( [ "$DISTRO_VERSION" -lt 20 ] && [ "$DISTRO_NAME" != "Ubuntu" ]);then
-      warning "tipi is designed to work with at least Ubuntu 20.04"
-      warning "But would you still like to try to install tipi (y/n) ?"
-      read answer
-      if [ "$answer" == "${answer#[Yy]}" ] ;then 
-       exit 1
+      if [ ! -f "/etc/arch-release" ]; then
```
Comment (pysco68) [inline install/install_for_macos_linux.sh:17]: I don't like that. Either we support only ubuntu or arch and ubuntu officially. In both cases that check is ugly. (like, why is it even nested?)

**Review by pysco68 — CHANGES_REQUESTED** [review]
Please un-nest that check for arch. I think its time we re-build that install script to properly support the select distributions we actually want to.

File: install/install_for_macos_linux.sh [hunk 7]
```diff
@@ -18,12 +37,14 @@ info() {
...
-should_install_unzip() {
-  if [[ $(command -v unzip) ]]; then
-    return 1
-  fi
+sucess() {
+  printf "\e[1;32m ---> $1 \e[0m \n"
 }
 
+if [ "$AVAIABLE_SIZE_FS" -le 10 ];then
+ warning "You may run out of space as you have only $AVAIABLE_SIZE_FS gb left on your tipi installation drive. A tipi installation typically requires between 2 and 10 gb of disk space depending on installation mode. You may change the installation location by setting the TIPI_HOME_DIR environment variable"
```
Comment (pysco68) [inline install/install_for_macos_linux.sh:45]: `gb` in lower case isn't any kind of known size notations. Please stick to correct Decimal or Binary notation of sizes (depending on what `df` returns - which is decimal *I think*)

File: install/install_for_macos_linux.sh [hunk 8]
```diff
@@ -1,14 +1,31 @@
...
+  printf "\e[1;33m ---> $1 \e[0m \n"
+}
  if [ "$(uname)" == "Linux" ]; then
     TIPI_URL="https://github.com/tipi-build/cli/releases/download/$VERSION/tipi-$VERSION-linux-x86_64.zip" 
+    AVAIABLE_SIZE_FS=$(df -H $HOME_DIR | awk '{ print $4}'  | cut -d'G' -f1 | cut -d'l' -f2 | tr -d '\n' )
+    DISTRO_NAME=$(cat /etc/*elease | grep DISTRIB_ID | cut -d '=' -f2)
+    DISTRO_VERSION=$(cat /etc/*elease | grep DISTRIB_RELEASE | cut -d '=' -f2 | sed 's@^[^0-9]*\([0-9]\+\).*@\1@')
+    ID_NAME = $(cat /etc/*elease | grep ^ID | cut -d '=' -f2)
+    if ([[ -z "$DISTRO_VERSION" ]] || [[ -z "$DISTRO_NAME" ]] || ( [ "$DISTRO_VERSION" -lt 20 ] || [ "$DISTRO_NAME" != "Ubuntu" ])) && ([[ -z "$ID_NAME" ]] || [ "$ID_NAME" != "arch" ] ); then
+      warning "tipi is currently supported on Ubuntu 20.04 or later only or on arch linux. You are running an unsuported distribution."
```
Comment (Lambourl) [inline install/install_for_macos_linux.sh:17]: explicit more with a variable is_ubuntu_20 and is_arch

**Comment by daminetreg** [comment]
superseeded by #36 that has been merged.

**Pull request closed without merging**